src/transformers/trainer.py

```diff
@@ -1089,6 +1090,11 @@ def get_optimizer_cls_and_kwargs(args: TrainingArguments) -> Tuple[Any, Any]:
                 optimizer_kwargs.update(bnb_kwargs)
             except ImportError:
                 raise ValueError("Trainer tried to instantiate bnb optimizer but bnb is not installed!")
+            if version.parse(importlib.metadata.version("bitsandbytes")) < version.parse("0.41.1"):
```

Comment:
I think you need to protect this check with the `is_bitsandbytes_available()` otherwise the Ci fails

Reply:
I think that it will first raise the ImportError above if `bitsandbytes` is not installed + using 8-bit optimizers.

Reply:
thx !